app/src/lib/stores/app-store.ts

```diff
@@ -987,17 +987,27 @@ export class AppStore extends TypedBaseStore<IAppState> {
         return this.currentMergeTreePromise
       }
 
-      if (tip.kind === TipState.Valid) {
+      if (tip.kind === TipState.Valid && aheadBehind.behind > 0) {
         const mergeTreePromise = promiseWithMinimumTimeout(
           () => mergeTree(repository, tip.branch, action.branch),
           500
-        ).then(mergeStatus => {
-          this.updateCompareState(repository, () => ({
-            mergeStatus,
-          }))
+        )
+          .catch(err => {
+            log.warn(
+              `Error occurred while trying to merge ${tip.branch.name} (${
+                tip.branch.tip.sha
+              }) and ${action.branch.name} (${action.branch.tip.sha})`,
+              err
+            )
+            return null
+          })
+          .then(mergeStatus => {
```

Comment:
Curious why this appears after the catch? Not that it makes much difference, but the typical ordering of these method calls is `then`, `catch`, then `finally`.

Reply:
The return value from `catch` is also a Promise, so this order of chaining means:

 - first, we run the async code that might error
 - then, if it errors, the `catch` callback allows us to handle the error and return something
 - finally, the `then` is a single callback to update the UI for both success and error cases

Flipping it around means that we might have to have UI code duplicated in `then` and `catch`